src/main/java/se/liu/ida/hefquin/engine/queryplan/physical/impl/PhysicalOpBindJoinWithUNION.java

```diff
@@ -0,0 +1,38 @@
+package se.liu.ida.hefquin.engine.queryplan.physical.impl;
+
+import se.liu.ida.hefquin.engine.federation.FederationMember;
+import se.liu.ida.hefquin.engine.federation.SPARQLEndpoint;
+import se.liu.ida.hefquin.engine.queryplan.ExpectedVariables;
+import se.liu.ida.hefquin.engine.queryplan.executable.impl.ops.ExecOpBindJoinSPARQLwithUNION;
+import se.liu.ida.hefquin.engine.queryplan.executable.impl.ops.UnaryExecutableOp;
+import se.liu.ida.hefquin.engine.queryplan.logical.UnaryLogicalOp;
+import se.liu.ida.hefquin.engine.queryplan.logical.impl.LogicalOpTPAdd;
+
+public class PhysicalOpBindJoinWithUNION extends BasePhysicalOpSingleInputJoin {
+
+	public PhysicalOpBindJoinWithUNION( final UnaryLogicalOp lop) {
+		super(lop);
+	}
+
+	@Override
+	public UnaryExecutableOp createExecOp(final ExpectedVariables... inputVars) {
+		for (final ExpectedVariables ev : inputVars) {
```

Comment:
You do not need to iterate here. The size of `inputVars` should be one. In fact, you should add an assert-check for that:
```
assert inputVars.length == 1;
```